src/auth/components/input/Input.tsx

```diff
@@ -6,6 +6,7 @@ import "./Input.scss"
 const Input = () => {
     const { username, setUsername } = useAppContext();
     const navigate = useNavigate();
+    const { role, setRole } = useAppContext();
```

Comment:
merge it with above call of context 
`const { username, setUsername, role, setRole } = useAppContext();`